lib/dataTransform.js

```diff
@@ -0,0 +1,114 @@
+const fs = require('fs'),
+  request = require('request'),
+  url = require('url'),
+  mime = require('mime'),
+  path = require('path'),
+  shell = require('shelljs');
+
+const download = (uri, filename) => {
+  return new Promise((resolve, reject) => {
+    request.head(uri, (err, res, body) => {
+      if (err) {
+        reject(err);
+      } else {
+        request(uri)
+          .pipe(fs.createWriteStream(filename))
+          .on('close', () => resolve(filename));
+      }
+    });
+  });
+};
+
+const filenameForUrl = uri => {
+  const filename = path.basename(url.parse(uri).pathname);
+  return `tmp/${filename}`;
+};
+
+const askForSlot = fileName => {
+  const params = { fileName: fileName, contentLength: fs.statSync(fileName)['size'], contentType: mime.getType(fileName) };
+  return new Promise((resolve, reject) => {
+    console.log(process.env.MARKETPLACE_TOKEN);
+    console.log(process.env.MARKETPLACE_URL);
+    console.log(DEPLOY_SERVICE_URL);
+    request.get(
+      {
+        url: `${DEPLOY_SERVICE_URL}/presign-url`,
+        extraHeaders: {
+          token: process.env.MARKETPLACE_TOKEN,
+          marketplace_domain: url.parse(process.env.MARKETPLACE_URL).hostname
+        },
+        qs: params,
+        json: true
+      },
+      (err, res, body) => {
+        if (err || res.statusCode != 200) {
+          console.log(res.body);
+          console.log(res.statusCode);
+          reject(err || body);
+        } else {
+          resolve(body.url);
+        }
+      }
+    );
+  });
+};
+
+const uploadFile = (fileName, s3Url) => {
+  var stats = fs.statSync(fileName);
+  return new Promise((resolve, reject) => {
+    fs.createReadStream(fileName).pipe(
+      request
+        .put({ url: s3Url, headers: { 'Content-Type': mime.getType(fileName), 'Content-Length': stats['size'] } })
+        .on('error', e => reject(e))
+        .on('end', () => resolve(s3Url))
+    );
+  });
+};
+
+const updateItem = (item, newUrl) => {
```

Comment:
Object.assign is not async action, why do you need to wrap it into promise? it can be just a function